pkg/ruler/namespace_protection.go

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package ruler
+
+import (
+	"context"
+	"errors"
+	"strings"
+)
+
+const (
+	OverrideProtectionHeader  = "X-Mimir-Ruler-Override-Namespace-Protection"
+	ProtectedNamespacesHeader = "X-Mimir-Ruler-Protected-Namespaces"
+)
+
+var (
+	ErrNoProtectionOverrideHeader = errors.New("no protection override header")
+	ErrNamespaceTargetNotMatch    = errors.New("namespace does not match target")
+)
+
+// IsNamespaceProtected returns true if the namespace is protected for the given user.
+func (r *Ruler) IsNamespaceProtected(userID string, namespace string) bool {
+	namespaces := r.limits.RulerProtectedNamespaces(userID)
+	for _, ns := range namespaces {
+		if namespace == ns {
+			return true
+		}
+	}
+
+	return false
+}
+
+// ProtectedNamespaceHeaderFromString returns a HTTPHeader with the given namespace as the value and ProtectedNamespacesHeader as the key.
+func ProtectedNamespaceHeaderFromString(namespace string) HTTPHeader {
```

Comment:
```suggestion
func ProtectedNamespacesHeaderFromString(namespace string) HTTPHeader {
```

Reply:
Fixed in https://github.com/grafana/mimir/pull/8444/commits/1c9c2770ffce6436b08c53d5ffe1e0178dba60a1